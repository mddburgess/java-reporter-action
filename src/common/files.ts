import * as glob from '@actions/glob';

export async function findFiles(searchPaths: string[]) {
    const globber = await glob.create(searchPaths.join('\n'));
    return await globber.glob();
}

export async function findFile(searchPath: string) {
    const paths = await findFiles([searchPath]);
    return paths[0];
}

export async function findRelativePath(searchPath: string) {
    const globber = await glob.create(searchPath);
    const searchRoot = globber.getSearchPaths()[0];
    const absolutePaths = await globber.glob();
    return absolutePaths[0] && absolutePaths[0].slice(searchRoot.length + 1);
}
