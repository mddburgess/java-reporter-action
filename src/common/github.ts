export interface Annotation {
    path: string,
    start_line: number,
    end_line: number,
    start_column?: number,
    end_column?: number,
    annotation_level: AnnotationLevel,
    title?: string,
    message: string,
    raw_details?: string
}

export type AnnotationLevel =
    | 'notice'
    | 'warning'
    | 'failure';
