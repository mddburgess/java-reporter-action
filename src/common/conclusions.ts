import { AnnotationSummary } from "../github/check-run";
import { CheckConclusion } from "../github/types";

const failureOnWarnings = (result: AnnotationSummary): CheckConclusion => {
  if (result.failures + result.warnings > 0) {
    return "failure";
  }
  return "success";
};

const neutralOnWarnings = (result: AnnotationSummary): CheckConclusion => {
  if (result.failures > 0) {
    return "failure";
  }
  if (result.warnings > 0) {
    return "neutral";
  }
  return "success";
}

export {
  failureOnWarnings,
  neutralOnWarnings
};
