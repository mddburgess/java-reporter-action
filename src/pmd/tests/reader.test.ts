import {findFile} from '../../common/files';
import PmdReportReader from '../reader';

describe('PMD report reader', () => {

    it('can read a PMD report', async () => {
        const reportPath = await findFile('**/pmd.xml');
        const report = new PmdReportReader().readReport(reportPath);
        expect(report).toMatchSnapshot();
    });
});
