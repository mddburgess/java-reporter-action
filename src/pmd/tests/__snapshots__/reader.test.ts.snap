// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PMD report reader can read a PMD report 1`] = `
Object {
  "violations": Array [
    Object {
      "endColumn": 20,
      "endLine": 3,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Avoid unused imports such as 'java.io.File'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 3,
    },
    Object {
      "endColumn": 41,
      "endLine": 11,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Do not hard code the IP address.",
      "priority": "3",
      "rule": "AvoidUsingHardCodedIP",
      "ruleset": "Best Practices",
      "startColumn": 31,
      "startLine": 11,
    },
    Object {
      "endColumn": 37,
      "endLine": 12,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Avoid unused private fields such as 'unusedPrivateField'.",
      "priority": "3",
      "rule": "UnusedPrivateField",
      "ruleset": "Best Practices",
      "startColumn": 20,
      "startLine": 12,
    },
    Object {
      "endColumn": 66,
      "endLine": 17,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.",
      "priority": "3",
      "rule": "CheckResultSet",
      "ruleset": "Best Practices",
      "startColumn": 9,
      "startLine": 17,
    },
    Object {
      "endColumn": 65,
      "endLine": 21,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Avoid unused method parameters such as 'unusedFormalParameter'.",
      "priority": "3",
      "rule": "UnusedFormalParameter",
      "ruleset": "Best Practices",
      "startColumn": 45,
      "startLine": 21,
    },
    Object {
      "endColumn": 66,
      "endLine": 21,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Avoid unused private methods such as 'unusedPrivateMethod(String)'.",
      "priority": "3",
      "rule": "UnusedPrivateMethod",
      "ruleset": "Best Practices",
      "startColumn": 18,
      "startLine": 21,
    },
    Object {
      "endColumn": 31,
      "endLine": 22,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/BestPractices.java",
      "message": "Avoid unused local variables such as 'unusedLocalVariable'.",
      "priority": "3",
      "rule": "UnusedLocalVariable",
      "ruleset": "Best Practices",
      "startColumn": 13,
      "startLine": 22,
    },
    Object {
      "endColumn": 2,
      "endLine": 59,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Too many static imports may lead to messy code.",
      "priority": "3",
      "rule": "TooManyStaticImports",
      "ruleset": "Code Style",
      "startColumn": 1,
      "startLine": 1,
    },
    Object {
      "endColumn": 24,
      "endLine": 3,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid importing anything from the package java.lang.",
      "priority": "4",
      "rule": "DontImportJavaLang",
      "ruleset": "Code Style",
      "startColumn": 1,
      "startLine": 3,
    },
    Object {
      "endColumn": 24,
      "endLine": 3,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.lang.String'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 3,
    },
    Object {
      "endColumn": 24,
      "endLine": 4,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid duplicate imports such as 'java.lang.String'.",
      "priority": "4",
      "rule": "DuplicateImports",
      "ruleset": "Code Style",
      "startColumn": 1,
      "startLine": 4,
    },
    Object {
      "endColumn": 24,
      "endLine": 4,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid importing anything from the package java.lang.",
      "priority": "4",
      "rule": "DontImportJavaLang",
      "ruleset": "Code Style",
      "startColumn": 1,
      "startLine": 4,
    },
    Object {
      "endColumn": 64,
      "endLine": 6,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.time.format.DateTimeFormatter.BASIC_ISO_DATE'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 6,
    },
    Object {
      "endColumn": 58,
      "endLine": 7,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.time.format.DateTimeFormatter.ISO_DATE'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 7,
    },
    Object {
      "endColumn": 63,
      "endLine": 8,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.time.format.DateTimeFormatter.ISO_DATE_TIME'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 8,
    },
    Object {
      "endColumn": 64,
      "endLine": 9,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.time.format.DateTimeFormatter.ISO_LOCAL_TIME'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 9,
    },
    Object {
      "endColumn": 61,
      "endLine": 10,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused imports such as 'java.time.format.DateTimeFormatter.ISO_INSTANT'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 10,
    },
    Object {
      "endColumn": 37,
      "endLine": 12,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "No need to explicitly extend Object.",
      "priority": "4",
      "rule": "ExtendsObject",
      "ruleset": "Code Style",
      "startColumn": 32,
      "startLine": 12,
    },
    Object {
      "endColumn": 9,
      "endLine": 17,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "This for loop could be simplified to a while loop.",
      "priority": "3",
      "rule": "ForLoopShouldBeWhileLoop",
      "ruleset": "Code Style",
      "startColumn": 9,
      "startLine": 15,
    },
    Object {
      "endColumn": 30,
      "endLine": 21,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused local variables such as 'value'.",
      "priority": "3",
      "rule": "UnusedLocalVariable",
      "ruleset": "Best Practices",
      "startColumn": 26,
      "startLine": 21,
    },
    Object {
      "endColumn": 24,
      "endLine": 21,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary use of fully qualified name 'java.lang.String' due to existing import 'java.lang.String'.",
      "priority": "4",
      "rule": "UnnecessaryFullyQualifiedName",
      "ruleset": "Code Style",
      "startColumn": 9,
      "startLine": 21,
    },
    Object {
      "endColumn": 5,
      "endLine": 26,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on interface 'UnnecessaryModifier': member interfaces are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 24,
    },
    Object {
      "endColumn": 5,
      "endLine": 30,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on enum 'UnnecessaryModifierEnum': nested enums are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 28,
    },
    Object {
      "endColumn": 15,
      "endLine": 33,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unnecessary return statements.",
      "priority": "3",
      "rule": "UnnecessaryReturn",
      "ruleset": "Code Style",
      "startColumn": 9,
      "startLine": 33,
    },
    Object {
      "endColumn": 17,
      "endLine": 37,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Useless parentheses.",
      "priority": "4",
      "rule": "UselessParentheses",
      "ruleset": "Code Style",
      "startColumn": 17,
      "startLine": 37,
    },
    Object {
      "endColumn": 26,
      "endLine": 41,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Avoid unused local variables such as 'instance'.",
      "priority": "3",
      "rule": "UnusedLocalVariable",
      "ruleset": "Best Practices",
      "startColumn": 19,
      "startLine": 41,
    },
    Object {
      "endColumn": 38,
      "endLine": 41,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Useless qualified this usage in the same class.",
      "priority": "3",
      "rule": "UselessQualifiedThis",
      "ruleset": "Code Style",
      "startColumn": 30,
      "startLine": 41,
    },
    Object {
      "endColumn": 42,
      "endLine": 47,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifiers 'public abstract' on method 'publicAbstract': the method is declared in an interface type.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 21,
      "startLine": 47,
    },
    Object {
      "endColumn": 50,
      "endLine": 48,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifiers 'public static final' on field 'publicStaticFinal': the field is declared in an interface type.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 25,
      "startLine": 48,
    },
    Object {
      "endColumn": 44,
      "endLine": 49,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'public' on class 'PublicStaticClass': members of interface types are implicitly public.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 49,
    },
    Object {
      "endColumn": 44,
      "endLine": 49,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on class 'PublicStaticClass': types nested within an interface type are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 49,
    },
    Object {
      "endColumn": 52,
      "endLine": 50,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'public' on interface 'PublicStaticInterface': members of interface types are implicitly public.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 50,
    },
    Object {
      "endColumn": 52,
      "endLine": 50,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on interface 'PublicStaticInterface': member interfaces are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 50,
    },
    Object {
      "endColumn": 41,
      "endLine": 55,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifiers 'public abstract' on method 'publicAbstract': the method is declared in an annotation type.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 21,
      "startLine": 55,
    },
    Object {
      "endColumn": 50,
      "endLine": 56,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifiers 'public static final' on field 'publicStaticFinal': the field is declared in an annotation type.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 25,
      "startLine": 56,
    },
    Object {
      "endColumn": 44,
      "endLine": 57,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'public' on class 'PublicStaticClass': members of annotation types are implicitly public.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 57,
    },
    Object {
      "endColumn": 44,
      "endLine": 57,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on class 'PublicStaticClass': types nested within an interface type are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 57,
    },
    Object {
      "endColumn": 52,
      "endLine": 58,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'public' on interface 'PublicStaticInterface': members of annotation types are implicitly public.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 58,
    },
    Object {
      "endColumn": 52,
      "endLine": 58,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/CodeStyle.java",
      "message": "Unnecessary modifier 'static' on interface 'PublicStaticInterface': member interfaces are implicitly static.",
      "priority": "3",
      "rule": "UnnecessaryModifier",
      "ruleset": "Code Style",
      "startColumn": 19,
      "startLine": 58,
    },
    Object {
      "endColumn": 13,
      "endLine": 9,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Design.java",
      "message": "These nested if statements could be combined.",
      "priority": "3",
      "rule": "CollapsibleIfStatements",
      "ruleset": "Design",
      "startColumn": 13,
      "startLine": 7,
    },
    Object {
      "endColumn": 43,
      "endLine": 14,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Design.java",
      "message": "Ternary operators that can be simplified with || or &amp;&amp;.",
      "priority": "3",
      "rule": "SimplifiedTernary",
      "ruleset": "Design",
      "startColumn": 16,
      "startLine": 14,
    },
    Object {
      "endColumn": 5,
      "endLine": 19,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Design.java",
      "message": "Overriding method merely calls super.",
      "priority": "3",
      "rule": "UselessOverridingMethod",
      "ruleset": "Design",
      "startColumn": 12,
      "startLine": 17,
    },
    Object {
      "endColumn": 37,
      "endLine": 9,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid unused imports such as 'org.example.pmd.BestPractices'.",
      "priority": "4",
      "rule": "UnusedImports",
      "ruleset": "Best Practices",
      "startColumn": 1,
      "startLine": 9,
    },
    Object {
      "endColumn": 37,
      "endLine": 9,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "No need to import a type that lives in the same package.",
      "priority": "3",
      "rule": "ImportFromSamePackage",
      "ruleset": "Error Prone",
      "startColumn": 1,
      "startLine": 9,
    },
    Object {
      "endColumn": 13,
      "endLine": 13,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Empty initializer was found.",
      "priority": "3",
      "rule": "EmptyInitializer",
      "ruleset": "Error Prone",
      "startColumn": 12,
      "startLine": 13,
    },
    Object {
      "endColumn": 18,
      "endLine": 20,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid using a branching statement as the last in a loop.",
      "priority": "2",
      "rule": "AvoidBranchingStatementAsLastInLoop",
      "ruleset": "Error Prone",
      "startColumn": 13,
      "startLine": 20,
    },
    Object {
      "endColumn": 36,
      "endLine": 25,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid creating BigDecimal with a decimal (float/double) literal. Use a String literal.",
      "priority": "3",
      "rule": "AvoidDecimalLiteralsInBigDecimalConstructor",
      "ruleset": "Error Prone",
      "startColumn": 16,
      "startLine": 25,
    },
    Object {
      "endColumn": 21,
      "endLine": 29,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Using multiple unary operators may be a bug, and/or is confusing.",
      "priority": "2",
      "rule": "AvoidMultipleUnaryOperators",
      "ruleset": "Error Prone",
      "startColumn": 16,
      "startLine": 29,
    },
    Object {
      "endColumn": 18,
      "endLine": 33,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Do not start a literal by 0 unless its an octal value.",
      "priority": "3",
      "rule": "AvoidUsingOctalValues",
      "ruleset": "Error Prone",
      "startColumn": 16,
      "startLine": 33,
    },
    Object {
      "endColumn": 9,
      "endLine": 39,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Method call on object which may be null.",
      "priority": "2",
      "rule": "BrokenNullCheck",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 37,
    },
    Object {
      "endColumn": 16,
      "endLine": 45,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Check the value returned by the skip() method of an InputStream to see if the requested number of bytes has been skipped.",
      "priority": "3",
      "rule": "CheckSkipResult",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 45,
    },
    Object {
      "endColumn": 58,
      "endLine": 51,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "This usage of the Collection.toArray() method will throw a ClassCastException.",
      "priority": "3",
      "rule": "ClassCastExceptionWithToArray",
      "ruleset": "Error Prone",
      "startColumn": 39,
      "startLine": 51,
    },
    Object {
      "endColumn": 18,
      "endLine": 55,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Dont use floating point for loop indices. If you must use floating point, use double.",
      "priority": "3",
      "rule": "DontUseFloatTypeForLoopIndices",
      "ruleset": "Error Prone",
      "startColumn": 14,
      "startLine": 55,
    },
    Object {
      "endColumn": 9,
      "endLine": 63,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty try blocks.",
      "priority": "3",
      "rule": "EmptyTryBlock",
      "ruleset": "Error Prone",
      "startColumn": 13,
      "startLine": 61,
    },
    Object {
      "endColumn": 9,
      "endLine": 65,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty catch blocks.",
      "priority": "3",
      "rule": "EmptyCatchBlock",
      "ruleset": "Error Prone",
      "startColumn": 11,
      "startLine": 63,
    },
    Object {
      "endColumn": 9,
      "endLine": 67,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty finally blocks.",
      "priority": "3",
      "rule": "EmptyFinallyBlock",
      "ruleset": "Error Prone",
      "startColumn": 11,
      "startLine": 65,
    },
    Object {
      "endColumn": 9,
      "endLine": 73,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty if statements.",
      "priority": "3",
      "rule": "EmptyIfStmt",
      "ruleset": "Error Prone",
      "startColumn": 24,
      "startLine": 71,
    },
    Object {
      "endColumn": 10,
      "endLine": 77,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty block statements.",
      "priority": "3",
      "rule": "EmptyStatementBlock",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 77,
    },
    Object {
      "endColumn": 9,
      "endLine": 81,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "An empty statement (semicolon) not part of a loop.",
      "priority": "3",
      "rule": "EmptyStatementNotInLoop",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 81,
    },
    Object {
      "endColumn": 9,
      "endLine": 88,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty switch statements.",
      "priority": "3",
      "rule": "EmptySwitchStatements",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 86,
    },
    Object {
      "endColumn": 9,
      "endLine": 94,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty synchronized blocks.",
      "priority": "3",
      "rule": "EmptySynchronizedBlock",
      "ruleset": "Error Prone",
      "startColumn": 29,
      "startLine": 92,
    },
    Object {
      "endColumn": 9,
      "endLine": 100,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid empty while statements.",
      "priority": "3",
      "rule": "EmptyWhileStmt",
      "ruleset": "Error Prone",
      "startColumn": 22,
      "startLine": 98,
    },
    Object {
      "endColumn": 13,
      "endLine": 107,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid modifying an outer loop incrementer in an inner loop for update expression.",
      "priority": "3",
      "rule": "JumbledIncrementer",
      "ruleset": "Error Prone",
      "startColumn": 13,
      "startLine": 105,
    },
    Object {
      "endColumn": 48,
      "endLine": 112,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "The null check here is misplaced; if the variable 'string' is null there will be a NullPointerException.",
      "priority": "3",
      "rule": "MisplacedNullCheck",
      "ruleset": "Error Prone",
      "startColumn": 43,
      "startLine": 112,
    },
    Object {
      "endColumn": 39,
      "endLine": 115,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Ensure you override both equals() and hashCode().",
      "priority": "3",
      "rule": "OverrideBothEqualsAndHashcode",
      "ruleset": "Error Prone",
      "startColumn": 20,
      "startLine": 115,
    },
    Object {
      "endColumn": 25,
      "endLine": 123,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid returning from a finally block.",
      "priority": "3",
      "rule": "ReturnFromFinallyBlock",
      "ruleset": "Error Prone",
      "startColumn": 13,
      "startLine": 123,
    },
    Object {
      "endColumn": 16,
      "endLine": 128,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Do not use if statements that are always true or always false.",
      "priority": "3",
      "rule": "UnconditionalIfStatement",
      "ruleset": "Error Prone",
      "startColumn": 13,
      "startLine": 128,
    },
    Object {
      "endColumn": 38,
      "endLine": 135,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Avoid unnecessary temporaries when converting primitives to Strings.",
      "priority": "3",
      "rule": "UnnecessaryConversionTemporary",
      "ruleset": "Error Prone",
      "startColumn": 30,
      "startLine": 135,
    },
    Object {
      "endColumn": 38,
      "endLine": 139,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "Invoke equals() on the object you've already ensured is not null.",
      "priority": "3",
      "rule": "UnusedNullCheckInEquals",
      "ruleset": "Error Prone",
      "startColumn": 38,
      "startLine": 139,
    },
    Object {
      "endColumn": 14,
      "endLine": 144,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/ErrorProne.java",
      "message": "An operation on an Immutable object (String, BigDecimal or BigInteger) wont change the object itself.",
      "priority": "3",
      "rule": "UselessOperationOnImmutable",
      "ruleset": "Error Prone",
      "startColumn": 9,
      "startLine": 144,
    },
    Object {
      "endColumn": 22,
      "endLine": 8,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Multithreading.java",
      "message": "Avoid unused local variables such as 'tg'.",
      "priority": "3",
      "rule": "UnusedLocalVariable",
      "ruleset": "Best Practices",
      "startColumn": 21,
      "startLine": 8,
    },
    Object {
      "endColumn": 40,
      "endLine": 8,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Multithreading.java",
      "message": "Avoid using java.lang.ThreadGroup; it is not thread safe.",
      "priority": "3",
      "rule": "AvoidThreadGroup",
      "ruleset": "Multithreading",
      "startColumn": 30,
      "startLine": 8,
    },
    Object {
      "endColumn": 26,
      "endLine": 12,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Multithreading.java",
      "message": "Dont call Thread.run() explicitly, use Thread.start().",
      "priority": "4",
      "rule": "DontCallThreadRun",
      "ruleset": "Multithreading",
      "startColumn": 9,
      "startLine": 12,
    },
    Object {
      "endColumn": 5,
      "endLine": 24,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Multithreading.java",
      "message": "Double checked locking is not thread safe in Java.",
      "priority": "1",
      "rule": "DoubleCheckedLocking",
      "ruleset": "Multithreading",
      "startColumn": 12,
      "startLine": 15,
    },
    Object {
      "endColumn": 34,
      "endLine": 8,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Performance.java",
      "message": "Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN).",
      "priority": "3",
      "rule": "BigIntegerInstantiation",
      "ruleset": "Performance",
      "startColumn": 16,
      "startLine": 8,
    },
    Object {
      "endColumn": 34,
      "endLine": 12,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/org/example/pmd/Performance.java",
      "message": "Avoid instantiating Boolean objects; reference Boolean.TRUE or Boolean.FALSE or call Boolean.valueOf() instead.",
      "priority": "2",
      "rule": "BooleanInstantiation",
      "ruleset": "Performance",
      "startColumn": 16,
      "startLine": 12,
    },
  ],
}
`;
